utils/nvme.go

```diff
@@ -160,170 +160,87 @@ func (n *Nvme) DriveCapabilities(ctx context.Context, logicalName string) ([]*co
-		})
+			Enabled:     (fna&0b100)>>2 != 0,
+		},
 	}
-
-	// nvme: cens
-	if (fna&0b10)>>1 == 0 {
-		caps = append(caps, &common.Capability{
-			Name:        "ceatsn",
-			Description: "Crypto Erase Applies to Single Namespace(s)",
-			Enabled:     false,
-		})
-	} else {
-		caps = append(caps, &common.Capability{
-			Name:        "ceatan",
-			Description: "Crypto Erase Applies to All Namespace(s)",
-			Enabled:     true,
-		})
-	}
-
-	// nvme: fmns
-	if (fna&0b1)>>0 == 0 {
-		caps = append(caps, &common.Capability{
-			Name:        "fatsn",
-			Description: "Format Applies to Single Namespace(s)",
-			Enabled:     false,
-		})
-	} else {
-		caps = append(caps, &common.Capability{
-			Name:        "fatan",
-			Description: "Format Applies to All Namespace(s)",
-			Enabled:     true,
-		})
-	}
-
-	return caps
 }
 
 // nolint:gomnd // bit twiddling and magic numbers go hand in hand for legibility
 func parseSanicap(sanicap uint) []*common.Capability {
-	caps := make([]*common.Capability, 0, 6)
-
-	caps = append(caps, &common.Capability{
-		Name:        "sanicap",
-		Description: "Sanitize Support",
-		Enabled:     sanicap != 0,
-	})
+	caps := []*common.Capability{
+		{
+			Name:        "sanicap",
+			Description: "Sanitize Support",
+			Enabled:     sanicap != 0,
+		},
+		{
+			Name:        "cer",
+			Description: "Crypto Erase Sanitize Operation Supported",
+			Enabled:     (sanicap&0b1)>>0 != 0,
+		},
+		{
+			Name:        "ber",
+			Description: "Block Erase Sanitize Operation Supported",
+			Enabled:     (sanicap&0b10)>>1 != 0,
+		},
+		{
+			Name:        "owr",
+			Description: "Overwrite Sanitize Operation Supported",
+			Enabled:     (sanicap&0b100)>>2 != 0,
+		},
+		{
+			Name:        "ndi",
+			Description: "No-Deallocate After Sanitize bit in Sanitize command Supported",
+			Enabled:     (sanicap&(0b1<<29))>>29 != 0,
+		},
+	}
 
 	// nvme: nodmmas
 	switch (sanicap & (0b11 << 30)) >> 30 {
 	case 0b00:
+		// nvme prints this out for 0b00:
+		//   "Additional media modification after sanitize operation completes successfully is not defined"
+		// So I'm taking "not defined" literally since we can't really represent 2 bits in a bool
+		// If we ever want this as a bool we could maybe call it "dmmas" maybe?
+	case 0b01, 0b10:
 		caps = append(caps, &common.Capability{
-			Name:        "ammasocsind",
-			Description: "Additional media modification after sanitize operation completes successfully is not defined",
-			Enabled:     false,
-		})
-	case 0b01:
-		caps = append(caps, &common.Capability{
-			Name:        "minamasocs",
-			Description: "Media is not additionally modified after sanitize operation completes successfully",
-			Enabled:     true,
-		})
-	case 0b10:
-		caps = append(caps, &common.Capability{
-			Name:        "miamasocs",
+			Name:        "nodmmas",
 			Description: "Media is additionally modified after sanitize operation completes successfully",
-			Enabled:     true,
+			Enabled:     (sanicap&(0b11<<30))>>30 == 0b10,
```

Comment:
please explain this shifting as well :)

Reply:
since you've included a link above, thats good enough for a point of reference, thanks!